src/init.cpp

```diff
@@ -203,15 +203,14 @@ void Shutdown(NodeContext& node)
     // Follow the lock order requirements:
     // * CheckForStaleTipAndEvictPeers locks cs_main before indirectly calling GetExtraFullOutboundCount
     //   which locks cs_vNodes.
-    // * ProcessMessage locks cs_main and g_cs_orphans before indirectly calling ForEachNode which
-    //   locks cs_vNodes.
-    // * CConnman::Stop calls DeleteNode, which calls FinalizeNode, which locks cs_main and calls
-    //   EraseOrphansFor, which locks g_cs_orphans.
+    // * ProcessMessage locks cs_main before indirectly calling ForEachNode which locks cs_vNodes.
+    // * CConnman::StopNodes grabs cs_vNodes lock which calls DeleteNode, which calls FinalizeNode,
+    //   which locks cs_main and calls EraseOrphansFor.
     //
-    // Thus the implicit locking order requirement is: (1) cs_main, (2) g_cs_orphans, (3) cs_vNodes.
+    // Thus the implicit locking order requirement is: (1) cs_main, (2) cs_vNodes.
     if (node.connman) {
         node.connman->StopThreads();
-        LOCK2(::cs_main, ::g_cs_orphans);
+        LOCK(::cs_main);
```

Comment:
I updated the comments which should still indicate why cs_main is needed there -- it needs to be locked before cs_vNodes, but cs_vNodes is locked first if you call StopThreads directly?

Reply:
Ah, very good point. And we can't take `cs_main` inside `StopNodes` because net isn't aware of `cs_main`.

It's outside the scope of this PR, but it'd be good to untangle this lock ordering dependency by eg making sure that cs_vNodes is not held when calling any PeerManager functions.

Reply:
See #21563

Reply:
21563 is merged